common/resource/fx.go

```diff
@@ -386,6 +388,7 @@ func RPCFactoryProvider(
 	svcName primitives.ServiceName,
 	logger log.Logger,
 	tlsConfigProvider encryption.TLSConfigProvider,
+	headersProvider sdk.HeadersProvider,
```

Comment:
If the header provider is going to be used for general purpose RPC and not just the SDK, I don't think it should be in the `sdk` package (even if it just so happens to _also_ match the SDK's interface). I think this should be somewhere where the SDK client and other gRPC clients can use it.

Reply:
The `HeadersProvider` interface is used only by the SDK client, while we create a `UnaryClientInterceptor` for the general RPC client.

If you have a suggestion for a better place for `HeadersProvider`, I don't mind moving it.

Reply:
> while we create a UnaryClientInterceptor for the general RPC client.

Right, so it is also being used for the interceptor for the general RPC client even though it's an interface placed in the `sdk` package. @dnr may have a better answer, but `common/rpc` or in this package seem reasonable.